### DISCUSSION TITLE
could not find native static library `mysqlclient`

### DISCUSSION THREAD

**Discussion by yk0n9** [diesel-rs/diesel#3224] [Q&A] [answered]
error: could not find native static library `mysqlclient`, perhaps an -L flag is missing?
error: could not compile `mysqlclient-sys` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `diesel_cli v1.4.1`, intermediate artifacts can be found at {Path}

OS: Windows 10
DB: MySQL
how should I fix it?

**Comment by weiznich** [comment]
This error means you need to provide a libmysqlclient version compatible to your compiler. Checkout the official mysql documentation for possible ways to install this library on your operating system.

**Reply by realtebo** [reply]
I am pretty sure you should document this step for new users instead of leaving them to google. Only an opinion

**Reply by weiznich** [reply]
@realtebo "just document it" is not an option here as you cannot document that for all possible ways users could setup their operating system. There is really just not much we can do here to improve the documentation other than stating that `libmysqlclient` is required and you need to install it.

**Reply by weiznich** [reply]
Otherwise if you have concrete suggestions: Please open a [PR here](https://github.com/sgrif/diesel.rs-website/), but please checkout existing (possibly closed) PR's on this topic first to be sure that you get the necessary context first.

**Comment by realtebo** [comment, marked as answer]
This is the connector

https://cdn.mysql.com/archives/mysql-connector-c/mysql-connector-c-6.1.11-winx64.zip

unzip it, for example in  C:\mysql-connector-c-6.1.11-winx64

then add windows env variable 

MYSQLCLIENT_LIB_DIR

pointing to

"C:\mysql-connector-c-6.1.11-winx64\lib\vs14"

then create a .env file in the project root, at same level of Cargo.toml

```
DATABASE_URL=mysql://user:password@localhost/dbname
MYSQLCLIENT_LIB_DIR="C:\mysql-connector-c-6.1.11-winx64\lib\vs14"

```
Then install from windows prompt (do not use prompt from vs code !)

    cargo install diesel_cli --no-default-features --features mysql

Ensure your .toml has this

```
[dependencies]
diesel = { version = "1.4.4", features = ["mysql"] }
dotenv = "0.15.0"
```
